### PR TITLE
98 feedback filtering

### PR DESCRIPTION
**Pull request by SanjinDedic**
The goal of this pull request is to improve game feedback for both game creators and users. Here are the benefits for both groups 

Game Creators:
- can create feedback in both markdown and dictionary form 
- can choose not to create feedback at all 
- Can create a single React component to customise the display of their feedback

Users:
- Feedback now available for all published simulations.
- Filtering feedback for specific players available for prisoners dilemma 

Here are the main changes to the system:
- feedback is now saved in the database and the SimulationResult table now has 2 new columns: 
feedback_str and feedback_json 
- publishing endpoint takes feedback as a parameter
- get_published_result is updated to include feedback from the database

### CODE REVIEW

File: backend/database.py [hunk 1]
```diff
@@ -318,14 +318,21 @@ def get_published_result(session, league_name):
                             table[value_name] = {}
                         table[value_name][result.team.name] = value
 
+            feedback = None
+            if sim.feedback_str is not None:
+                feedback = sim.feedback_str
+            elif sim.feedback_json is not None:
+                feedback = json.loads(sim.feedback_json)
```
Comment (abaruchi) [inline backend/database.py:325]: Usually, when I'm using `if` / `elif` I prefer to use an `else` at the end (python doesn't have the case / switch, but it has the elif, which has the same behaviour, and usually, case / switch has a default value). My suggestion here:

```
            if sim.feedback_str is not None:
                feedback = sim.feedback_str
            elif sim.feedback_json is not None:
                feedback = json.loads(sim.feedback_json)
            else:
                feedback = None
```
It makes more clear that... if the previous conditions arent satisfied, the default value will be None.

File: backend/database.py [hunk 2]
```diff
@@ -352,13 +359,12 @@ def process_simulation_results(sim, league_name, active=None):
...
-            raise SimulationResultFormatError(f"Table item '{key}' is not a dictionary")
-        
-        if set(value.keys()) != set(total_points.keys()):
-            raise SimulationResultFormatError(f"Keys in table item '{key}' do not match keys in total_points")
+    # Add feedback handling
+    feedback = None
+    if sim.feedback_str is not None:
+        feedback = sim.feedback_str
+    elif sim.feedback_json is not None:
+        feedback = json.loads(sim.feedback_json)
```
Comment (abaruchi) [inline backend/database.py:367]: Same here (https://github.com/SanjinDedic/agent_games/pull/113/commits/ad20ba1280a58c560c35009085b60a270e04e813#r1875035914)

File: backend/dockerfile_script.log
Comment (abaruchi) [inline backend/dockerfile_script.log:1]: Do we need the docker log file being versioned? If not, please add to the `.gitignore` file.

File: backend/database.py (hunk 1 above)
Comment (SanjinDedic) [reply backend/database.py:325]: Makes sense, imlemented

File: backend/database.py (hunk 2 above)
Comment (SanjinDedic) [reply backend/database.py:367]: implemented

File: backend/dockerfile_script.log
Comment (SanjinDedic) [reply backend/dockerfile_script.log:1]: done

**Comment by SanjinDedic** [comment]
@abaruchi Anything else to to review? ready to merge?

**Pull request merged**